coreax/proposed_data.py

```diff
@@ -0,0 +1,96 @@
+# © Crown Copyright GCHQ
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Classes for reading unsupervised and supervised input data.
+
+In order to calculate a coreset, :meth:`~coreax.reduction.Coreset.fit` requires an
+instance of a subclass of :class:`Data`. It is necessary to use
+:class:`Data` because :class:`~coreax.reduction.Coreset` requires a
+two-dimensional :class:`~jax.Array`. Data reductions are performed along the first
+dimension.
+"""
+
+from __future__ import annotations
+
+import equinox as eqx
+import jax.numpy as jnp
+from jaxtyping import Array, Shaped
+
+
+# pylint: disable=too-few-public-methods
+class Data(eqx.Module):
+    r"""
+    Class for representing unsupervised data.
+
+    :param data: An :math:`n \times d` array defining the unsupervised dataset
+    :param weights: An :math:`n`-vector of weights with. Each element of the weights
```

Comment:
Typo here, same in the Supervised data class. Should probably be "where each element" instead of "with. Each element"

Reply:
Fixed